scripts/check-redact-log

```diff
@@ -1,9 +1,21 @@
 #!/usr/bin/env bash
 # This script checks if there is plain (not redacted) log in TiKV.
+set -euo pipefail
 
-if grep -r -n -P '(?<!hex_)encode_upper' -C 1 --color=always --include \*.rs --exclude hex.rs --exclude-dir target . ; then
-    echo "To print user data into info logs or error messages, use log_wrappers::Value() instead of hex::encode_upper. The former will respect \`security.redact-info-log\` config and filter out user data from info log if needed. Otherwise, use \`log_wrappers::hex_encode_upper\` to get around the lint error. See https://github.com/tikv/tikv/pull/9250 for more information." 
-    exit 1
+function error_msg() {
+        echo "To print user data into info logs or error messages, use log_wrappers::Value() instead of hex::encode_upper. The former will respect \`security.redact-info-log\` config and filter out user data from info log if needed. Otherwise, use \`log_wrappers::hex_encode_upper\` to get around the lint error. See https://github.com/tikv/tikv/pull/9250 for more information." >&2
+}
+
+if command -v uname >/dev/null && [[ "$(uname)" == "Darwin" ]] ; then
```

Comment:
What's the purpose of `command -v uname`?

Reply:
I am used to checking for the existence of an executable. Probably not needed in this case so I am happy to remove it.

Reply:
Please keep it simple.